ibis/backends/tests/sql/test_sql.py

```diff
@@ -461,16 +461,20 @@ def test3():
 
 
 def test_gh_1045(test1, test2, test3, snapshot):
```

Comment:
This test is changing, and thus the API is breaking here.

In the original issue the bug was an incorrect column pruning optimization.

The existing code doesn't work anymore, because the semantics of projections are clarified:

You can project out of a table using a parent table, only if any preceding projections still contain references used in the final projection.

That's a mouthful, so let's look at an example:

```python
t = ibis.table({"a": "int", "b": "string"})
```

This is allowed:

```python
t.mutate(x=t.a).select(t.b)  # t.mutate(x=t.a) still contains `t.b`
```

This is not allowed:

```python
t.drop("b").select(t.b)  # b no longer exists so you can't use it in a subsequent projection
```

I'll adjust the test to reflect this behavior.

Reply:
**UPDATE:**

We *might* be able to relax this later.

Reply:
We need to accumulate these examples in the future docs since now we have clear semantics about the inputs we accept.